templates/master/01-master-kubelet/baremetal/units/kubelet.yaml

```diff
@@ -0,0 +1,38 @@
+name: "kubelet.service"
+enabled: true
+contents: |
+  [Unit]
+  Description=Kubernetes Kubelet
+  Wants=rpc-statd.service
+
+  [Service]
+  Type=notify
+  ExecStartPre=/bin/mkdir --parents /etc/kubernetes/manifests
+  ExecStartPre=/bin/rm -f /var/lib/kubelet/cpu_manager_state
+  EnvironmentFile=/etc/os-release
+  EnvironmentFile=-/etc/kubernetes/kubelet-workaround
+  EnvironmentFile=-/etc/kubernetes/kubelet-env
+
+  ExecStart=/usr/bin/hyperkube \
+      kubelet \
+        --config=/etc/kubernetes/kubelet.conf \
+        --bootstrap-kubeconfig=/etc/kubernetes/kubeconfig \
+        --rotate-certificates \
+        --kubeconfig=/var/lib/kubelet/kubeconfig \
+        --container-runtime=remote \
+        --container-runtime-endpoint=/var/run/crio/crio.sock \
+        --allow-privileged \
+        --node-labels=node-role.kubernetes.io/master,node.openshift.io/os_version=${VERSION_ID},node.openshift.io/os_id=${ID} \
```

Comment:
personal opinion only, feel free to ignore.

Do you have a strong reason why this platform is deviating from other platforms of OpenShift.

This deviation is not the best way forward. It adds another dimension of things operators need to know for this platform as implicit API

A better solution would be to teach operators to explicitly tolerate that taint for this platform.

Reply:
> Do you have a strong reason why this platform is deviating from other platforms of OpenShift.

For KNI, the hardware has a starting point of around 40 cores and 256GB of ram each. It's a big expense to dedicate that much hardware only to the control plane.  A larger cluster could be willing to have that much overhead, but our minimum is 4 servers, which means having 75% overhead.

Could we make the NoSchedule taint configurable through the installer for all platforms, and avoid the special case for baremetal?

> A better solution would be to teach operators to explicitly tolerate that taint for this platform.

How many operators would that be? It seems preferable to me to do it in one place, rather than many. Also: would users need to be aware of this if they wanted to run their own workloads on the baremetal platform? What if they're using operators we don't control?

Reply:
I actually don't think this should be the default for the baremetal platform.  We should default to the same as any other platform.

What's needed is a way to declaratively specify that workloads should be able to run on masters.  We have a product under development that would use this by default, but that's different from the more general IPI baremetal support.  There's a 4.2 epic for this "workloads on masters" support.

In short, this part should probably be dropped for now, and wait for the fully supported way to specify this.

Reply:
I very much agree with https://github.com/openshift/machine-config-operator/pull/722#discussion_r283352069 - we want a `controlPlane: schedulable` bool or something.
See https://github.com/openshift/machine-config-operator/pull/722#issuecomment-492828738 and following.